sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/DefaultOfferRequirementProvider.java

```diff
@@ -173,22 +173,38 @@ private ExecutorRequirement getExecutorRequirement(
             String serviceName,
             UUID targetConfigurationId) throws InvalidRequirementException {
         List<Protos.TaskInfo> podTasks = TaskUtils.getPodTasks(podInstance, stateStore);
+        Protos.ExecutorInfo executorInfo = null;
 
         for (Protos.TaskInfo taskInfo : podTasks) {
             Optional<Protos.TaskStatus> taskStatusOptional = stateStore.fetchStatus(taskInfo.getName());
             if (taskStatusOptional.isPresent()
                     && taskStatusOptional.get().getState() == Protos.TaskState.TASK_RUNNING) {
+                executorInfo = taskInfo.getExecutor();
                 LOGGER.info(
                         "Reusing executor from task '{}': {}",
                         taskInfo.getName(),
-                        TextFormat.shortDebugString(taskInfo.getExecutor()));
-                return ExecutorRequirement.create(taskInfo.getExecutor());
+                        TextFormat.shortDebugString(executorInfo));
             }
         }
 
+        Map<String, Protos.Resource> volumeMap = new HashMap<>();
+        if (executorInfo == null) {
```

Comment:
I do think that `ExecutorInfo` should continue to be passed to `ExecutorRequirement` -- `DefaultOfferRequirementProvider` has to know a lot of things to create an `ExecutorInfo`, like the pod name, how to construct the `ContainerInfo`, the pod user, URIs...right now I don't think it makes sense to pass all of that stuff to an `ExecutorRequirement` just so it can know how to create the protobuf. All of the data dependencies here could probably stand some refactoring, but it's a larger process than just moving where a proto is instantiated.